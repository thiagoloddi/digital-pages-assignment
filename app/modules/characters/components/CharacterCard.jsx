import React from 'react';
import Paper from '@material-ui/core/Paper';
import injectSheet from 'react-jss';
import Image from '../../../components/ui/Image';

const style = {
  paper: {
    marginTop: '20px',
    padding: '15px',
    '& > *': {
      display: 'inline-block'
    }
  },
  image: {
    height: '250px',
    width: '250px',
  },
  heading: {
    margin: 0
  },
  info: {
    verticalAlign: 'top',
    paddingLeft: '20px',
    width: 'calc(100% - 250px)'
  }
}

const CharacterCard = ({ name, thumbnail, description, classes }) => {
  return (
    <Paper className={classes.paper}>
      <Image className={classes.image} src={`${thumbnail.path}.${thumbnail.extension}`} />
      <div className={classes.info}>
        <h1 className={classes.heading}>{name}</h1>
        <p>{description}</p>
      </div>
    </Paper>
  )
}

export default injectSheet(style)(CharacterCard);