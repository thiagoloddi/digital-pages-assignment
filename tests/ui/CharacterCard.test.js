import React from 'react';
import renderer from 'react-test-renderer';
import CharacterCard from '../../app/modules/characters/components/CharacterCard';

const createNodeMock = e => {
  if(e.type == 'img') {
    return {};
  }
}

it('renders correctly', () => {
  const tree = renderer
    .create(<CharacterCard name="Char name" thumbnail={{ path: "path", ext: ".jpg" }} description="A description" />, { createNodeMock })
    .toJSON();
  expect(tree).toMatchSnapshot();
});