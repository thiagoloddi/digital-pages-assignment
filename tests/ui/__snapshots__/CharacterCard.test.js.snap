// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 CharacterCard-paper-0-2-1"
>
  <div
    className="Image-container-0-2-5 CharacterCard-image-0-2-2"
  >
    <img
      className="Image-img-0-2-6"
      src="path.undefined"
      style={
        Object {
          "visibility": "hidden ",
        }
      }
    />
    <img
      className="Image-loading-0-2-7"
      src="/assets/progress.gif"
      style={null}
    />
  </div>
  <div
    className="CharacterCard-info-0-2-4"
  >
    <h1
      className="CharacterCard-heading-0-2-3"
    >
      Char name
    </h1>
    <p>
      A description
    </p>
  </div>
</div>
`;
